Mathlib/Geometry/Manifold/Instances/Sphere.lean

```diff
@@ -223,23 +223,34 @@ theorem stereo_left_inv (hv : ‖v‖ = 1) {x : sphere (0 : E) 1} (hx : (x : E)
   have pythag : 1 = a ^ 2 + ‖y‖ ^ 2 := by
     have hvy' : ⟪a • v, y⟫_ℝ = 0 := by simp only [inner_smul_left, hvy, mul_zero]
     convert norm_add_sq_eq_norm_sq_add_norm_sq_of_inner_eq_zero _ _ hvy' using 2
-    · simp [← split]
+    -- porting note (#10745): was simp [← split] but wasn't finding `norm_eq_of_mem_sphere`
+    · simp only [norm_eq_of_mem_sphere, Nat.cast_one, mul_one, ← split]
     · simp [norm_smul, hv, ← sq, sq_abs]
     · exact sq _
+  -- Porting note: added to work around cancel_denoms and nlinarith failures
+  have duh : ‖y.val‖ ^ 2 = 1 - a ^ 2 := by
+    rw [← Submodule.coe_norm, pythag]; ring
   -- two facts which will be helpful for clearing denominators in the main calculation
   have ha : 1 - a ≠ 0 := by
     have : a < 1 := (inner_lt_one_iff_real_of_norm_one hv (by simp)).mpr hx.symm
     linarith
   -- the core of the problem is these two algebraic identities:
   have h₁ : (2 ^ 2 / (1 - a) ^ 2 * ‖y‖ ^ 2 + 4)⁻¹ * 4 * (2 / (1 - a)) = 1 := by
-    field_simp; simp only [Submodule.coe_norm] at *; nlinarith
+    -- Porting note: used to be `field_simp; simp only [Submodule.coe_norm] at *; nlinarith`
```

Comment:
Does this need a mathlib issue?

Reply:
Sorry, this diff is junk, I'll start over on this file. :-(